core/src/main/java/com/linecorp/armeria/client/RefreshingAddressResolver.java

```diff
@@ -149,7 +149,25 @@ private void sendQueries(List<DnsQuestion> questions, String hostname,
         final Future<List<DnsRecord>> recordsFuture = resolver.sendQueries(questions, hostname);
         recordsFuture.addListener(f -> {
             if (!f.isSuccess()) {
-                result.complete(new CacheEntry(null, -1, questions, f.cause()));
+                final Throwable cause = f.cause();
+
+                // In Netty, only cache if the failure was not because of an IO error / timeout that was
+                // caused by the query itself. To figure out that, we need to check the cause of the
+                // UnknownHostException. If it's null, then we can cache the cause.
+                // However, this is very vulnerable because Netty can change the behavior while we are not
+                // noticing that. So sending a PR to upstream would be the best solution.
```

Comment:
Would it be possible to at least file an issue now and link it here so not forgotten?

Reply:
Yes, let me make one. 😃